Podcast/DiscoverViewController.swift

```diff
@@ -124,91 +78,44 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
     
     func tableView(_ tableView: UITableView,
                             cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "DiscoverTableViewCellIdentifier") as! DiscoverTableViewCell
-        cell.episode = feedEpisodes[indexPath.row]
-        cell.layoutSubviews()
+        let cell = tableView.dequeueReusableCell(withIdentifier: "FeedTableViewCellIdentifier") as! FeedTableViewCell
+        cell.delegate = self
+        cell.setupWithCard(card: feedEpisodes[indexPath.row])
         return cell
     }
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         
-        guard let cell = feedTableView.cellForRow(at: indexPath) as? DiscoverTableViewCell else { return }
-
-        cell.isExpanded = !cell.isExpanded
-
-        tableView.beginUpdates()
-        tableView.reloadData()
-        tableView.endUpdates()
+        guard let cell = feedTableView.cellForRow(at: indexPath) as? FeedTableViewCell else { return }
         
     }
-    
-    //MARK: -
-    //MARK: CollectionView DataSource
-    //MARK: -
-    
 
-    func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
-    }
-    
-    func collectionView(_ collectionView: UICollectionView,
-                                 numberOfItemsInSection section: Int) -> Int {
-        return categories.count
-    }
     
-    func collectionView(_ collectionView: UICollectionView,
-                                 cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "CategoriesCollectionViewIdentifier",
-                                                      for: indexPath) as! CategoriesCollectionViewCell
-        cell.categoryName = categories[indexPath.item]
-        cell.layer.cornerRadius = 2.0
-        cell.layer.masksToBounds = true
-        return cell
-    }
+    //MARK: -
+    //MARK: FeedTableViewCell Delegate
+    //MARK: - 
     
-    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        print("selected \(categories[indexPath.item])")
-        let vc = CategoriesFeedViewController()
-        vc.category = categories[indexPath.item]
-        navigationController!.pushViewController(vc, animated: false)
+    func feedTableViewCellDidPressRecommendButton(feedTableViewCell: FeedTableViewCell) {
+        
+        guard let cardIndexPath = feedTableView.indexPath(for: feedTableViewCell), let card = feedEpisodes[cardIndexPath.row] as? EpisodeCard else { return }
+        
+        card.isRecommended = !card.isRecommended
+        feedTableViewCell.didPressRecommendedButtonChangeView(isRecommended: card.isRecommended)
```

Comment:
should probably rename this to feedTableViewCell.setRecommendButtonToState(isRecommended: card.isRecommended)

that function should also set the card.isRecommended property instead of doing that in the previous line here